### PR TITLE
docs: reorganize index and readme

### PR DESCRIPTION
**Pull request by itaysk**
## Description
I wanted to update the docs homepage as it was out of date and I took the opportunity to make some more improvements. The most notable change is that I tried to unify the content of the readme and the docs index so that it's easier to keep both updated in the future (there are still differences in the layout but the content can be shared).

A few of things that crossed my mind but I didn't implement. I can implement if you agree, or we can leave those out of this PR:
1. using inline md links (`[description](link)`) would make writing much easier. with footnotes style you need to scroll up and down while writing which is very distracting.
2. Not sure we need the highlights in the readme/index, as we're moving very fast and it's hard to maintain.
3. I know we discussed it many times but still not sure "CLI" section makes sense. not sure what I suggest ("Docs"? "Reference Docs"?) just throwing this thought out there.

## Related PRs
- [ ] #3025 - the ecosystem link assume it was merged.

Remove this section if you don't have related PRs.

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by knqyf263** [comment]
>1. using inline md links ([description](link)) would make writing much easier. with footnotes style you need to scroll up and down while writing which is very distracting.

We used to use inline links, but someone suggested the footnotes style and we eventually switched to that because links are long and make it hard to read the document description on our editor. Also, it is easy to find all the links and we can update one place if the same link is used several times. As for relative paths, I'm ok with inline links.

File: docs/index.md [hunk 1]
```diff
@@ -2,33 +2,72 @@
...
-Trivy has different scanners that look for different security issues, and different targets where it can find those issues.
+Trivy is a comprehensive and versatile security scanner. Trivy has *scanners* that look for security issues, and *targets* where it can find those issues.
 
-Targets:
+Targets (what Trivy can scan):
 
 - Container Image
 - Filesystem
 - Git repository (remote)
-- Kubernetes cluster or resource
```
Comment (AnaisUrlichs) [inline docs/index.md:17]: Maybe we want to point out both separately -- Trivy scans running Kubernetes workloads inside the cluster but also static Kubernetes manifests

**Review by AnaisUrlichs — COMMENTED** [review]
Added one minor comment

File: docs/index.md (hunk 1 above)
Comment (itaysk) [reply docs/index.md:17]: > workloads inside the cluster but also static Kubernetes manifests

by "static manifests" you mean k8s yaml on disk? in this case, it a fs target no? or you meant scanning a resource in k8s, in this case I didn't get the contrast with "kubernetes workloads" in your suggestion.

File: docs/index.md [hunk 2]
```diff
@@ -37,18 +76,81 @@ Please see [LICENSE][license] for Trivy licensing information.
...
 
-[integrations]: ./tutorials/integrations/index.md
-[slack]: https://slack.aquasec.com
-[license]:  https://github.com/aquasecurity/trivy/blob/main/LICENSE
\ No newline at end of file
+</details>
+
+## Highlights
+
+- Comprehensive vulnerability detection
```
[the remark this answers is not in the document]
Comment (itaysk) [reply docs/index.md:113]: I don't understsand, why 4 spaces?

File: README.md [hunk 3]
```diff
@@ -5,56 +5,59 @@
...
 [![License: Apache-2.0][license-img]][license]
-[![GitHub All Releases][github-all-releases-img]][release]
+[![GitHub Downloads][github-downloads-img]][release]
 ![Docker Pulls][docker-pulls]
 
 [📖 Documentation][docs]
 </div>
 
-Trivy (`tri` pronounced like **tri**gger, `vy` pronounced like en**vy**) is a comprehensive security scanner. It is reliable, fast, extremely easy to use, and it works wherever you need it.
+Trivy is a comprehensive and versatile security scanner. Trivy has *scanners* that look for security issues, and *targets* where it can find those issues.
```
[the remark this answers is not in the document]
Comment (itaysk) [reply README.md:14]: WDYT of this solution?

File: docs/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (itaysk) [reply docs/index.md:113]: I'll fix anyway

File: docs/index.md (hunk 1 above)
Comment (AnaisUrlichs) [reply docs/index.md:17]: I mean that the 'trivy config' command can check Kubernetes manifests and that the 'k8s' command can check running workloads

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (knqyf263) [reply README.md]: LGTM

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**